p2p/src/types.rs

```diff
@@ -223,18 +223,18 @@ pub struct P2PConfig {
 	pub seeding_type: Seeding,
 
 	/// The list of seed nodes, if using Seeding as a seed type
-	pub seeds: Option<Vec<PeerAddr>>,
+	pub seeds: Option<Vec<String>>,
 
 	/// Capabilities expose by this node, also conditions which other peers this
 	/// node will have an affinity toward when connection.
 	pub capabilities: Capabilities,
 
-	pub peers_allow: Option<Vec<PeerAddr>>,
+	pub peers_allow: Option<Vec<String>>,
```

Comment:
It reduces type safety, can we somehow keep it while supporting the new functionality? I haven't checked the code carefully, but perhaps smth like implementing TryFrom/TryInto would help to hide the logic of conversion.

Reply:
Could potentially implement a Serde deserializer for Vec<PeerAddr>. But it wouldn't be just deserializing it would be resolving the DNS names as well. And there are potentially multiple IP addresses for each DNS names which complicates things. I can give it a go.

Reply:
Outdated, resolved.